clang/lib/AST/Interp/Interp.h

```diff
@@ -1437,7 +1437,7 @@ bool OffsetHelper(InterpState &S, CodePtr OpPC, const T &Offset,
     return false;
   };
 
-  unsigned MaxOffset = MaxIndex - Ptr.getIndex();
+  T MaxOffset = T::from(MaxIndex - Ptr.getIndex(), Offset.bitWidth());
```

Comment:
Should MaxIndex just be a `T` as well?  I wonder if this function should just be 'fully dedicated' to being `T` instead of just this 1 variable here?

Reply:
Dunno, We never compare `MaxIndex` with anything of type `T`. It's even only used twice and once is in the diagnostic output. The other one is line 1440 above.